eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/bag/sorted/mutable/SynchronizedMutableSortedBagMultimap.java

```diff
@@ -0,0 +1,210 @@
+import org.eclipse.collections.api.block.function.Function2;
+import org.eclipse.collections.api.block.predicate.Predicate2;
+import org.eclipse.collections.api.multimap.Multimap;
+import org.eclipse.collections.api.multimap.bag.MutableBagMultimap;
+import org.eclipse.collections.api.multimap.list.MutableListMultimap;
+import org.eclipse.collections.api.multimap.sortedbag.ImmutableSortedBagMultimap;
+import org.eclipse.collections.api.multimap.sortedbag.MutableSortedBagMultimap;
+import org.eclipse.collections.api.tuple.Pair;
+import org.eclipse.collections.impl.factory.Multimaps;
+import org.eclipse.collections.impl.multimap.AbstractSynchronizedMutableMultimap;
+import org.eclipse.collections.impl.multimap.SynchronizedMultimapSerializationProxy;
+
+public class SynchronizedMutableSortedBagMultimap<K, V>
+        extends AbstractSynchronizedMutableMultimap<K, V>
+        implements MutableSortedBagMultimap<K, V>, Serializable
+{
+    private static final long serialVersionUID = 1L;
+
+    public SynchronizedMutableSortedBagMultimap(MutableSortedBagMultimap<K, V> newMultiMap)
+    {
+        super(newMultiMap);
+    }
+
+    public SynchronizedMutableSortedBagMultimap(MutableSortedBagMultimap<K, V> newMultimap, Object newLock)
+    {
+        super(newMultimap, newLock);
+    }
+
+    protected Object writeReplace()
+    {
+        return new SynchronizedMultimapSerializationProxy<K, V>(this.getDelegate());
+    }
+
+    /**
+     * This method will take a Multimap and wrap it directly in a SynchronizedMutableMultimap.
+     */
+    public static <K, V, M extends Multimap<K, V>> SynchronizedMutableSortedBagMultimap<K, V> of(M multimap)
+    {
+        if (multimap == null)
+        {
+            throw new IllegalArgumentException("cannot create a SynchronizedMutableMultimap for null");
+        }
+
+        return new SynchronizedMutableSortedBagMultimap<K, V>(Multimaps.mutable.sortedBag.withAll(multimap));
+    }
+
+    /**
+     * This method will take a Multimap and wrap it directly in a SynchronizedMutableMap.
+     * Additionally, a developer specifies which lock to use with the collection.
+     */
+    public static <K, V, M extends Multimap<K, V>> SynchronizedMutableSortedBagMultimap<K, V> of(M multimap, Object lock)
+    {
+        if (multimap == null)
+        {
+            throw new IllegalArgumentException("cannot create a SynchronizedMutableMultimap for null");
+        }
+
+        return new SynchronizedMutableSortedBagMultimap<K, V>(Multimaps.mutable.sortedBag.withAll(multimap), lock);
+    }
+
+    public MutableSortedBag<V> replaceValues(K key, Iterable<? extends V> values)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().replaceValues(key, values);
+        }
+    }
+
+    public MutableSortedBag<V> removeAll(Object key)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().removeAll(key);
+        }
+    }
+
+    public void clear()
```

Comment:
I think a method like `clear()` which is on `MutableMultimap` could be pulled up higher in the inheritance hierarchy.